xarray/core/dataset.py

```diff
@@ -2528,9 +2531,15 @@ def to_zarr(
             Any additional parameters for the storage backend (ignored for local
             paths).
         zarr_version : int or None, optional
-            The desired zarr spec version to target (currently 2 or 3). The
-            default of None will attempt to determine the zarr version from
-            ``store`` when possible, otherwise defaulting to 2.
+
+            .. deprecated:: 2024.9.1
+            Use ``zarr_format`` instead.
+
+        zarr_format : int or None, optional
```

Comment:
Will need to add a note below about how `fill_value` is interpreted differently depending on the format.